restli-tools/src/main/resources/apiVmTemplates/rest.create_id.vm

```diff
@@ -0,0 +1,59 @@
+#*
+   Copyright (c) 2021 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*#
+
+  #doc($method.schema.doc)
+  public CompletionStage<${spec.keyClassName}> create(
+      ${spec.entityClassName} entity#if( $method.hasParams()),#end
+      #methodParams($method)
+      ) {
+    #paramsRequestMap($method)
+    @SuppressWarnings("unchecked")
+    IdResponseDecoder<${spec.keyClassName}> idResponseDecoder = new IdResponseDecoder<>(
+        (TypeSpec<${spec.keyClassName}>) _resourceSpec.getKeyType(),
+        _resourceSpec.getKeyParts(),
+        _resourceSpec.getComplexKeyType());
+    CreateIdRequest<${spec.keyClassName}, ${spec.entityClassName}> request = new CreateIdRequest<>(
+        entity,
+        Collections.emptyMap(),
+        Collections.emptyList(),
+        idResponseDecoder,
+        _resourceSpec,
+        queryParams,
+        queryParamClasses,
+        ORIGINAL_RESOURCE_PATH,
+        Collections.emptyMap(),
+        RestliRequestOptions.DEFAULT_OPTIONS,
+        null);
+    CompletableFuture<${spec.keyClassName}> result = new CompletableFuture<>();
+    Task<Void> responseTask = _client.createTask(request).transform("Task to completion stage",
+        responseTry -> {
+          if (responseTry.isFailed()) {
+            result.completeExceptionally(responseTry.getError());
+          } else {
+            Response<IdResponse<${spec.keyClassName}>> response = responseTry.get();
+            if (response.hasError()) {
+              result.completeExceptionally(response.getError());
+            } else {
+              result.complete(response.getEntity().getId());
+            }
+          }
+          return Success.of((Void) null);
+        });
+    _engine.run(responseTask);
```

Comment:
I saw you left comment regarding "scheduleAndRun" draft and said start a new plan if no context found.

I think you are talking about this case?
```java
<completionStageImpl>.andThen( () -> {
<resource>.get(); //try to reuse the existing context, if no context found, start a new plan
})
```
But I see here this line (Rest.Li) framework is the right place to run the _engine, therefore I didn't reply to your comment in that draft.

Do you think we can catch the exception and run a plan if no plan found?

Reply:
Throwing -> catching an exception during a normal execution flow is not good (it is also less performant). 

We should atleast make scheduleAndRun return a boolean indicating if the task was scheduled for execution. 
After that we can do the check followed by engine.run here. 

However, adding that to engine itself might be useful for other users. And this logic of checking scheduleAndRun followed by engine.run need not be duplicated everywhere. wdyt?

Reply:
the problem is that the task wouldn't have context regarding which engine runs it.

In the https://github.com/linkedin/parseq/pull/291 I added a API to take engine as a parameter.

Reply:
What I meant is Engine::scheduleAndRun(task)

Reply:
What is the behavior difference between `Engine::scheduleAndRun(task)` and `task.scheduleOrRun(engine)` proposed?

Also I will also be interested to know what is the use case of
Engine::scheduleAndRun(task)